### DISCUSSION TITLE
Code highlighting for HTTP requests

### DISCUSSION THREAD

**Discussion by Lars418** [PrismJS/prism#2936] [Q&A] [answered]
Github also has code higlighting. Using `http` after the three backticks allows you to format routes, e.g.

```http
POST /user/login/
```

Prism also has `http` as a supported language, but it only seems to highlight qualified urls (e.g. `https://github.com/)`. Is there a way to get the same result with prism like the github syntax highlighting for routes?

**Comment by RunDevelopment** [comment, marked as answer]
HTTP requires a valid HTTP version in each [request line](https://datatracker.ietf.org/doc/html/rfc7230#section-3.1.1), so your snippet isn't a valid HTTP request. Highlighting [works as expected with a version](https://prismjs.com/test.html#language=http&text=POST%20%2Fuser%2Flogin%2F%20HTTP%2F1.1).

**Reply by Lars418** [reply]
Thanks for the quick reply, good to know.

**Reply by dandv** [reply]
We want to highlight HTTP requests with complex parameters ([examples](https://weaviate.io/developers/weaviate/api/rest/objects)). Adding the HTTP version would be irrelevant to our readers. How can we highlight the requests without adding `HTTP/1.1`? (we currently abuse `js`)

Also, the `http` highlighter doesn't seem to highlight key/value URL parameters:

<img width="516" alt="image" src="https://user-images.githubusercontent.com/33569/220088831-46e7881b-58a2-44e0-af14-f93547156ff9.png">
